File: testSuite/test-inactiveNumerics.pl
#!/usr/bin/env perl
use strict;
use warnings;
use PDL;
use PDL::NiceSlice;
use PDL::IO::HDF5;

# Run models that test that inactive properties are evolved when using an ODE solver that does not support inactive property evaluation.
# Andrew Benson (04-May-2022)

# Make output directory.
system("mkdir -p outputs/");

# Run the model.
system("cd ..; ./Galacticus.exe testSuite/parameters/inactiveNumerics.xml");
unless ( $? == 0 ) {
    print "FAIL: inactiveNumerics model failed to run\n";
}

# Check that luminosities are non-zero.
my $nonZeroLuminosities = 0;
my $model               = new PDL::IO::HDF5("outputs/inactiveNumerics.hdf5");
my $nodes               = $model->group('Outputs/Output1/nodeData');
foreach my $datasetName ( $nodes->datasets() ) {
    next
	unless ( $datasetName =~ m/Luminosities/ );
    my $dataset = $nodes->dataset($datasetName)->get();
    $nonZeroLuminosities = 1
	if ( any($dataset > 0.0) );
}
if ( $nonZeroLuminosities ) {
    print "SUCCESS: non-zero luminosities are present\n";
} else {
    print "FAIL: all luminosities are zero\n";
}

exit;

File: testSuite/test-inactiveNumerics.py
#!/usr/bin/env python3
import subprocess
import sys
import h5py
import numpy as np

# Run models that test that inactive properties are evolved when using an ODE solver
# that does not support inactive property evaluation.
# Andrew Benson (ported to Python)

# Make output directory.
subprocess.run("mkdir -p outputs/", shell=True)

# Run the model.
status = subprocess.run("cd ..; ./Galacticus.exe testSuite/parameters/inactiveNumerics.xml", shell=True)
if status.returncode != 0:
    print("FAIL: inactiveNumerics model failed to run")

# Check that luminosities are non-zero.
nonZeroLuminosities = False
with h5py.File("outputs/inactiveNumerics.hdf5", "r") as model:
    nodeData = model["Outputs/Output1/nodeData"]
    for datasetName in nodeData.keys():
        if "Luminosities" not in datasetName:
            continue
        dataset = nodeData[datasetName][:]
        if np.any(dataset > 0.0):
            nonZeroLuminosities = True
            break

if nonZeroLuminosities:
    print("SUCCESS: non-zero luminosities are present")
else:
    print("FAIL: all luminosities are zero")
